toolkit/scripts/toolchain.mk

```diff
@@ -297,3 +305,23 @@ $(toolchain_rpms): $(TOOLCHAIN_MANIFEST) $(depend_REBUILD_TOOLCHAIN)
 		$(call print_error,\nToolchain download failed. See above errors for more details.) \
 	}
 endif
+
+# ./out/RPMS is reserved for RPMs generated by the tooling, all other RPMs are cached in the ./build folder. If REBUILD_TOOLCHAIN=y is set
+# the toolchain RPMs should also be placed into out/RPMs.
+
+# ./out/SRPMS are still handled by build_official_toolchain_rpms.sh as a side effect of building the toolchain tar.gz.
+ifeq ($(REBUILD_TOOLCHAIN),y)
+$(RPMS_DIR): $(toolchain_out_rpms)
+
+# For each toolchain rpm in ./out/RPMS, add a dependency on its counterpart in the normal toolchain directory
+$(foreach toolchain_out_rpm,$(toolchain_out_rpms),$(eval $(toolchain_out_rpm): $(filter %/$(notdir $(toolchain_out_rpm)),$(toolchain_rpms))))
```

Comment:
This is insanely hard to read, took me a while to understand what's going on. But if we're going to rebuilt the toolchain, shouldn't we always expect `$(toolchain_out_rpms)` to depend on the whole set of `$(toolchain_rpms)`? Something along these lines:
```make
$(toolchain_out_rpms): $(toolchain_rpms)
```

Reply:
Or I'm completely misunderstanding this and the following lines and how they work together. I think I understand what we want to do, just not how this these lines work and get us there.

Reply:
I front loaded the filter operation to the makefile foreach so I could do `cp $^ $@` (ie `cp src.rpm dest.rpm`). I can implement that inside the bash instead if that is more readable, or at the lest I can add more to the comment.

Reply:
Ok, I think I've found a more readable way to do this.
```makefile
# For each toolchain rpm in ./out/RPMS, add a dependency on the counterparts in the normal toolchain directory:
# Each path in $(toolchain_out_rpms) corresponds to a .rpm file we expect to have been built by the toolchain target and made available in ./out/RPMS.
# Those rpms however are placed by default in ./build/toolchain/* (listed in $(toolchain_rpms)). So if we want a copy placed in ./out/RPMS 
# we will need to copy it over. We can filter the list of toolchain rpms $(toolchain_rpms) to find the source that matches the target ($@),
# then copy it over.
$(toolchain_out_rpms): $(toolchain_rpms)
	@src_rpm='$(filter %/$(notdir $@),$(toolchain_rpms))'  && \
	if [ ! -f "$@" \
			-o "$$src_rpm" -nt "$@" ] ; then \
		echo "Placing built toolchain RPM $(notdir $@) into $(RPMS_DIR)" && \
		cp $$src_rpm $@; \
	fi || $(call print_error, Failed to duplicate '$$src_rpm' to '$@' )
else
$(toolchain_out_rpms):
	@$(call print_error, Cannot build toolchain rpm ($@) unless REBUILD_TOOLCHAIN=y is set)
endif
```

Reply:
I still use the make pre-processor since the grep command ended up being quite a mess.